Tests/Marketplace/search_and_install_packs.py

```diff
@@ -692,51 +729,67 @@ def search_and_install_packs_and_their_dependencies_private(test_pack_path: str,
     installs them.
+
     Args:
         pack_ids (list): A list of the pack ids to search and install.
         client (demisto_client): The client to connect to.
         hostname (str): Hostname of instance. Using for logs.
-        install_packs_one_by_one(bool): Whether to install packs in small batches.
-            If false - install all packs in one batch.
-
+        multithreading (bool): Whether to use multithreading to install packs in parallel.
+            If multithreading is used, installation requests will be sent in batches of each pack and its dependencies.
+        is_post_update (bool): Whether the installation is in post update mode. Defaults to False.
     Returns (list, bool):
         A list of the installed packs' ids, or an empty list if is_nightly == True.
         A flag that indicates if the operation succeeded or not.
     """
     host = hostname or client.api_client.configuration.host
 
-    logging.info(f'Starting to search packs in server: {host}')
+    logging.info(f'Starting search for packs to install on: {host}')
 
-    packs_to_install: list = []  # we save all the packs we want to install, to avoid duplications
-    installation_request_body: list = []  # the packs to install, in the request format
-    batch_packs_install_request_body: list = []    # list of lists of packs to install if install packs one by one.
+    packs_to_install: list = []  # Packs we want to install, to avoid duplications
+    installation_request_body: list = []  # Packs to install, in the request format
+    batch_packs_install_request_body: list = []  # List of lists of packs to install if not using multithreading .
     # Each list contain one pack and its dependencies.
-    packs_in_the_list_to_install: list = []    # list of packs that are already in the list to install.
+    collected_dependencies: list = []  # List of packs that are already in the list to install.
+    master_commit_hash = os.getenv("LAST_UPLOAD_COMMIT")
+
+    if not master_commit_hash:
+        raise ValueError("Required 'LAST_UPLOAD_COMMIT' environment variable is missing.")
 
     lock = Lock()
 
-    if install_packs_one_by_one:
+    kwargs = {
+        'client': client,
+        'packs_to_install': packs_to_install,
+        'installation_request_body': installation_request_body,
+        'lock': lock,
+        'collected_dependencies': collected_dependencies,
+        'is_post_update': is_post_update,
+        'multithreading': multithreading,
+        'batch_packs_install_request_body': batch_packs_install_request_body,
+        'commit_hash': master_commit_hash,
+    }
+
+    if is_post_update:
+        logging.info("Detected post-update run mode. "
+                     "Pack deprecation status will be determined using local pack metadata.")
+
+    else:
+        logging.info("Detected pre-update run mode. "
+                     "Pack deprecation status will be determined using Marketplace API.")
```

Comment:
Remove from here, if you want to log this, add this to the `is_pack_deprecated` function, and use `logging.debug` not info. 
```suggestion
```